python/sglang/srt/layers/dp_attention.py

```diff
@@ -355,16 +355,27 @@ def disable_dp_size():
 
 
 def get_dp_local_info(forward_batch: ForwardBatch) -> Tuple[torch.Tensor, torch.Tensor]:
-    # `get_dp_local_info` is only called in global DP gather and scatter. We use global DP rank here.
+    # Compute local token range for current DP rank.
+    # For LogitsMetadata, use global_num_tokens_for_logprob_gpu (pruned tokens needing logits).
+    # For ForwardBatch (MLP/Attention), use global_num_tokens_gpu (all tokens in batch).
     dp_rank = get_attention_dp_rank()
 
     if forward_batch.dp_local_start_pos is None:
-        cumtokens = torch.cumsum(forward_batch.global_num_tokens_gpu, dim=0)
+        # Select metadata source based on context
+        if (
```

Comment:
How can `dp_local_start_pos` be None in `LogitsMetadata`? It is expected to be processed here: https://github.com/sgl-project/sglang/blob/0bfa394affbf62db60390411294ae6c67cb24455/python/sglang/srt/layers/logits_processor.py#L577.

`LogitsMetadata` has a separate processing logic. We should keep it separated. Perhaps we should add assertion like `assert isinstance(forward_batch, ForwardBatch)` to avoid misuse.